src/EditorFeatures/Core/Shared/Preview/PreviewSolutionCrawlerRegistrationService.cs

```diff
@@ -68,6 +74,34 @@ public void Unregister(Workspace workspace, bool blockingShutdown = false)
             {
                 source.Cancel();
             }
+
+            ClearDiagnostics(workspace);
+
+            AssertCleanup();
+        }
+
+        private void ClearDiagnostics(Workspace workspace)
+        {
+            var analyzer = _provider.CreateIncrementalAnalyzer(workspace);
+            var args = _service.GetDiagnosticsUpdatedEventArgs(workspace, projectId: null, documentId: null, cancellationToken: CancellationToken.None);
```

Comment:
this service know exactly what service it is getting back. it is IDiagnosticAnalyzerService since it get IIncrementalAnalyzerProvider from it. 

that act basically shows it knows what it is getting. also, not sure how things are cached or whether it is cached need to be exposed outside.